src/main/kotlin/com/todo/deoji/infrastructure/global/jwt/adapter/ParseTokenAdapter.kt

```
@@ -0,0 +1,69 @@
+
+        return try {
+            Jwts.parserBuilder()
+                .setSigningKey(secret)
+                .build()
+                .parseClaimsJws(token)
+        } catch (e: Exception) {
+            when (e) {
+                is InvalidClaimException -> throw TokenNotValidException()
+                is ExpiredJwtException -> throw ExpiredTokenException()
+                is JwtException -> throw TokenNotValidException()
+                else -> throw RuntimeException()
```

detekt (1.23.7)

[warning] A call to the default constructor of an exception was detected. Instead one of the constructor overloads should be called. This allows to provide more meaningful exceptions.

(detekt.exceptions.ThrowingExceptionsWithoutMessageOrCause)